src/Avalonia.Controls.DataGrid/DataGridDataConnection.cs

```diff
@@ -379,17 +379,36 @@ public object GetDataItem(int index)
 
         public bool GetPropertyIsReadOnly(string propertyName)
         {
-            if (DataType != null)
+            Type propertyType = DataType;
+            if (propertyType != null)
             {
+                if (propertyType.GetIsReadOnly())
+                    return true;
+
                 if (!String.IsNullOrEmpty(propertyName))
                 {
-                    Type propertyType = DataType;
+                    Type? propertyReflectableType = null;
+                    if (typeof(IReflectableType).IsAssignableFrom(propertyType))
+                    {
+                        IEnumerator en = DataSource.GetEnumerator();
```

Comment:
You are expected to try casting enumerator to IDisposable by IEnumerable.GetEnumerator convention